code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm

```diff
@@ -57,6 +64,65 @@
 	adjacent_turfs.Cut()
 	return ..()
 
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/add_filters(filter_or_filters)
+	if(!islist(filter_or_filters))
+		filter_or_filters = list(filter_or_filters)
+
+	for(var/gas_to_filter in filter_or_filters)
+		var/translated_gas = istext(gas_to_filter) ? gas_id2path(gas_to_filter) : gas_to_filter
+
+		if(ispath(translated_gas, /datum/gas))
+			filter_types |= translated_gas
+			continue
+
+	var/turf/open/our_turf = get_turf(src)
+	var/datum/gas_mixture/turf_gas = our_turf?.air
+	if(!our_turf || !turf_gas)
+		return FALSE
+
+	check_atmos_process(src, turf_gas, turf_gas.temperature)
+	return TRUE
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/remove_filters(filter_or_filters)
+	if(!islist(filter_or_filters))
+		filter_or_filters = list(filter_or_filters)
+
+	for(var/gas_to_filter in filter_or_filters)
+		var/translated_gas = istext(gas_to_filter) ? gas_id2path(gas_to_filter) : gas_to_filter
+
+		if(ispath(translated_gas, /datum/gas))
+			filter_types -= translated_gas
+			continue
+
+	var/turf/open/our_turf = get_turf(src)
+	var/datum/gas_mixture/turf_gas = our_turf?.air
+	if(!our_turf || !turf_gas)
+		return FALSE
+
+	check_atmos_process(src, turf_gas, turf_gas.temperature)
```

Comment:
source is the turf isn't it?

Reply:
no

Reply:
shouldn't it be? that's what the signal sends

Reply:
i am dumb nm